semgrep-core/src/parsing/Parse_pattern.ml

```diff
@@ -98,8 +98,21 @@ let parse_pattern lang ?(print_errors = false) str =
     | Lang.Ts
     | Lang.Js
     | Lang.Vue ->
-        let any = Parse_js.any_of_string str in
-        Js_to_generic.any any
+        (* This is a simplified version of run_either in Parse_target.ml. We
+         * don't need most of the logic there when we're parsing patterns, so it
+         * doesn't make sense to reuse it. However, we should abstract this out
+         * if we start using fallback parsers for patterns in other languages.
+         * *)
+        let js_ast =
+          try Parse_js.any_of_string str with
+          | exn -> (
```

Comment:
You need to add 
          | Timeout _ as e -> Exception.catch_and_reraise e
otherwise you could swallow a Timeout.
We should write a semgrep rule for that, but we probably need to improve the try/with semgrep pattern for OCaml.

Reply:
https://github.com/returntocorp/semgrep/pull/5999